bootstrap/kubeadm/api/v1alpha4/kubeadmbootstrapconfig_types.go

```diff
@@ -95,6 +98,31 @@ type KubeadmConfigSpec struct {
 	// For more information, refer to https://github.com/kubernetes-sigs/cluster-api/pull/2763#discussion_r397306055.
 	// +optional
 	UseExperimentalRetryJoin bool `json:"useExperimentalRetryJoin,omitempty"`
+
+	// IgnitionConfig contains Ignition specific configuration.
+	//
+	// If set, bootstrap data will be generated in Ignition format instead of cloud-init.
+	//
+	// +optional
+	IgnitionConfig *IgnitionConfig `json:"ignitionConfig,omitempty"`
```

Comment:
```suggestion
	Ignition *IgnitionSpec `json:"ignition,omitempty"`
```

There is a lot of `Config` suffixes

Reply:
I agree there are a lot of `Config` suffixes but this one might be worth keeping since Ignition (the binary and associated version) != Ignition config specification (and version). I.e. Ignition v0.x supports config spec v2.x, and Ignition v2.x supports config spec v3.x.
I know it's already confusing, which is why I'd be explicit here.

Reply:
OTOH, I'd expect a struct called `IgnitionConfig` to contain an Ignition config.  This struct contains higher-level configs that _transpile_ into Ignition configs.

Reply:
right! `IgnitionTranspilerConfig` maybe?

Reply:
>right! IgnitionTranspilerConfig maybe?

It could also contain a `Version` field to control what version of Ignition is generated in the future. `IgnitionTranspilerConfig` name wouldn't make sense anymore.

Reply:
@vincepri can we mark this as resolved?